### DISCUSSION TITLE
Text alignment doesn't work

### DISCUSSION THREAD

**Discussion by Nordsoft91** [axmolengine/axmol#1561] [Q&A] [answered]
- axmol version: axmol-2.1.0-rc7
- devices test on: MacOS 12.6
- developing environments
   - cmake version: 3.27.2
Steps to Reproduce:
1. In MainScene:init() create a label (store pointer as class member)
```
label = ax::Label::createWithTTF("", "fonts/arial.ttf", 12);
label->setPosition({50, 600});
label->setMaxLineWidth(200);
label->setAlignment(ax::TextHAlignment::LEFT);
addChild(label);
```
2. on mouse move (enable event dispatcher)
```
void MainScene::onMouseMove(ax::Event* event)
{
	ax::EventMouse* e = static_cast<ax::EventMouse*>(event);
	label->setString(ax::StringUtils::format("X:%f\nY:%f", e->getCursorX(), e->getCursorY()));
}
```
3. Start game and move mouse to the left outside window

Expected result: text is aligner to the left, adding more characters doesn't move text

Actual result: when mouse is 100 pix or more outside screen, text is shifting

Note: steps above can be modified, bug appears with any kind of text

https://github.com/axmolengine/axmol/assets/9308612/644c93d0-b7ed-43e1-919b-2712cb01dd89

**Comment by DelinWorks** [comment]
use `setAnchorPoint` instead of setting the text alignment, text alignment should really be used with setDimension

**Comment by rh101** [comment, marked as answer]
@Nordsoft91 As @DelinWorks pointed out, to correctly align text, the label would need to have fixed dimensions.  The reason it's moving in your example is simply because the dimensions of the box change as the string contents are updated, and given it's set to anchor point of `Vec2::ANCHOR_MIDDLE` by default, you can see how that would affect the left alignment of text.

If you have no other concerns related to this, then please close this issue.

**Comment by Nordsoft91** [comment]
thanks, with anchor (0, 1) works as naturally expected
